### PR TITLE
MINOR: Small cleanups/refactoring in kafka.controller

### PR DESCRIPTION
**Pull request by mimaison**
- Updated logging to use string templates
- Minor refactors
- Fixed a few typos

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by asfgit** [comment]
FAILURE
 7973 tests run, 5 skipped, 1 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8081 tests run, 5 skipped, 0 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8081 tests run, 5 skipped, 0 failed.
--none--

File: core/src/main/scala/kafka/controller/PartitionStateMachine.scala [hunk 1]
```diff
@@ -417,9 +417,8 @@ class PartitionStateMachine(config: KafkaConfig,
 
   private def logInvalidTransition(partition: TopicPartition, targetState: PartitionState): Unit = {
     val currState = partitionState(partition)
-    val e = new IllegalStateException("Partition %s should be in the %s states before moving to %s state"
-      .format(partition, targetState.validPreviousStates.mkString(","), targetState) + ". Instead it is in %s state"
-      .format(currState))
+    val e = new IllegalStateException(s"Partition $partition should be in the ${targetState.validPreviousStates.mkString(",")}" +
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/PartitionStateMachine.scala:420]: Space missing at the end.

File: core/src/main/scala/kafka/controller/ReplicaStateMachine.scala [hunk 2]
```diff
@@ -381,16 +381,15 @@ class ReplicaStateMachine(config: KafkaConfig,
...
+      s"states before moving to $targetState state. Instead it is in $currState state")
     logFailedStateChange(replica, currState, targetState, e)
   }
 
   private def logFailedStateChange(replica: PartitionAndReplica, currState: ReplicaState, targetState: ReplicaState, t: Throwable): Unit = {
     stateChangeLogger.withControllerEpoch(controllerContext.epoch)
-      .error("Controller %d epoch %d initiated state change of replica %d for partition [%s,%d] from %s to %s failed"
-      .format(controllerId, controllerContext.epoch, replica.replica, replica.topic, replica.partition, currState, targetState), t)
+      .error(s"Controller $controllerId epoch ${controllerContext.epoch} initiated state change of replica ${replica.replica} " +
+        s"for partition [${replica.topic},${replica.partition}] from $currState to $targetState failed", t)
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/ReplicaStateMachine.scala:392]: We should use `replica.topicPartition` here (existing issue).

File: core/src/main/scala/kafka/controller/KafkaController.scala [hunk 3]
```diff
@@ -1361,8 +1360,7 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
         val partitions = zkClient.getPreferredReplicaElection
         val partitionsForTopicsToBeDeleted = partitions.filter(p => topicDeletionManager.isTopicQueuedUpForDeletion(p.topic))
         if (partitionsForTopicsToBeDeleted.nonEmpty) {
-          error("Skipping preferred replica election for partitions %s since the respective topics are being deleted"
-            .format(partitionsForTopicsToBeDeleted))
+          error(s"Skipping preferred replica election for partitions $partitionsForTopicsToBeDeleted since the respective topics are being deleted")
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/KafkaController.scala:1363]: Nit: long line.

File: core/src/main/scala/kafka/controller/KafkaController.scala [hunk 4]
```diff
@@ -1193,8 +1193,7 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
       controllerContext.partitionReplicaAssignment = controllerContext.partitionReplicaAssignment.filter(p =>
         !deletedTopics.contains(p._1.topic))
       controllerContext.partitionReplicaAssignment ++= addedPartitionReplicaAssignment
-      info("New topics: [%s], deleted topics: [%s], new partition replica assignment [%s]".format(newTopics,
-        deletedTopics, addedPartitionReplicaAssignment))
+      info(s"New topics: [$newTopics], deleted topics: [$deletedTopics], new partition replica assignment [$addedPartitionReplicaAssignment]")
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/KafkaController.scala:1196]: Nit: long line.

File: core/src/main/scala/kafka/controller/KafkaController.scala [hunk 5]
```diff
@@ -879,8 +878,8 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
           val controllerEpoch = leaderIsrAndControllerEpoch.controllerEpoch
           if (controllerEpoch > epoch)
             throw new StateChangeFailedException("Leader and isr path written by another controller. This probably" +
-              "means the current controller with epoch %d went through a soft failure and another ".format(epoch) +
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/KafkaController.scala:882]: Space missing after `probably` (existing issue).

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM. I noticed a few nits (noted below) and I'll fix them before merging to trunk.

**Comment by mimaison** [comment]
Thanks !

**Pull request closed without merging**